NumLib/NamedFunctionProvider.h

```diff
@@ -0,0 +1,32 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef NUMLIB_NAMEDFUNCTIONPROVIDER_H
+#define NUMLIB_NAMEDFUNCTIONPROVIDER_H
+
+#include "NamedFunction.h"
+
+namespace NumLib
+{
+
+//! Interface used for providing named functions.
+class NamedFunctionProvider
+{
+public:
+    virtual std::vector<NamedFunction> getNamedFunctions() const
```

Comment:
This interface is meant to be pulled into many different classes, e.g. into the interfaces of (all) material properties. If one does not provide any named function and because there is a default implementation one does not need to bother in a concrete implementation to implement this method again. So I'd like to keep it as it is.